sematic/examples/testing_pipeline/BUILD

```diff
@@ -22,6 +22,8 @@ sematic_pipeline(
         requirement("ray"),
     ],
     deps = [
+        requirement("debugpy"),
+        requirement("ray"),
```

Comment:
The PR description says that until now dependencies had to be added to both the layers and to the library dependencies. Both my experience and the before state of this example contradict this. Please clarify.

Are the dependencies duplicated here in order to run the example locally without building the image? Why not automatically pull image layer dependencies into the library dependencies when running locally instead of having to duplicate these declarations?

Reply:
sorry, the wording of the PR description was not very clear; I reworded it to make it more clear what's going on. The "before" behavior was that if you added anything to "layers" it would automatically also be included as a dependency of the py binary. This is something that you may not want--you might have a dependency which is only relevant as a cloud image layer, but not for local execution (ex: if the dependency is something that people locally will have installed out-of-band of bazel). We have a user who has a use case where they want something to be a dependency only in the cloud. That was not possible before this PR. The behavior *now* is that you can add a dependency in "layers" to make it get used as an image cached layer, but if you want it to also be available in the binary you need to explicitly include it in the "deps" field. So:

deps => all dependencies that are needed for both local and cloud. Guaranteed to be included for both local and cloud
image_layers => purely used in the image, likely exclusively used to control caching behavior

Reply:
Also modified the docstring to clarify

Reply:
Thank you for clarifying! This is a breaking change for existing pipelines. Please add a breaking change note to the changelog, with instructions on how to fix existing pipelines.

Reply:
done